### PR TITLE
Adapt Blob API to sync CallApi of OlpClient

### PR DESCRIPTION
**Pull request by diachenko-mischa**
Adapt implementation to use sync CallApi of OlpClient. Fix tests to use
fixed Blob api. Fix code style.

Relates-To: OLPEDGE-1168
Signed-off-by: Diachenko Mykahilo <ext-mykhailo.z.diachenko@here.com>

### CODE REVIEW

File: olp-cpp-sdk-dataservice-read/src/generated/api/BlobApi.cpp [hunk 1]
```diff
@@ -31,46 +31,37 @@ namespace dataservice {
...
+                                       const std::string& layer_id,
+                                       const std::string& data_handle,
+                                       boost::optional<std::string> billing_tag,
+                                       boost::optional<std::string> range,
+                                       client::CancellationContext context) {
+  std::multimap<std::string, std::string> header_params;
+  header_params.insert(std::make_pair("Accept", "application/json"));
   if (range) {
-    headerParams.insert(std::make_pair("Range", *range));
+    header_params.insert(std::make_pair("Range", *range));
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/src/generated/api/BlobApi.cpp]:
```suggestion
    header_params.emplace("Range", *range);
```

File: olp-cpp-sdk-dataservice-read/src/generated/api/BlobApi.cpp [hunk 2]
```diff
@@ -31,46 +31,37 @@ namespace dataservice {
...
   }
 
-  std::multimap<std::string, std::string> queryParams;
-  if (billingTag) {
-    queryParams.insert(std::make_pair("billingTag", *billingTag));
+  std::multimap<std::string, std::string> query_params;
+  if (billing_tag) {
+    query_params.insert(std::make_pair("billingTag", *billing_tag));
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/src/generated/api/BlobApi.cpp]:
```suggestion
    query_params.emplace("billingTag", *billing_tag);
```

File: olp-cpp-sdk-dataservice-read/src/generated/api/BlobApi.cpp [hunk 3]
```diff
@@ -31,46 +31,37 @@ namespace dataservice {
...
   }
 
-  std::multimap<std::string, std::string> formParams;
+  std::multimap<std::string, std::string> form_params;
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/src/generated/api/BlobApi.cpp]: Do we need to create an unused var? Instead, I would just pass a default constructed rvalue:

```cpp
client.CallApi(metadata_uri, "GET", query_params, header_params,
                     {}, nullptr, "", context);
```

File: olp-cpp-sdk-dataservice-read/src/generated/api/BlobApi.cpp [hunk 4]
```diff
@@ -31,46 +31,37 @@ namespace dataservice {
...
-        } else {
-          dataResponseCallback(
-              // TODO: response from HttpResponse should be already in
-              // raw data format to avoid copy
-              std::make_shared<std::vector<unsigned char>>(str_response.begin(),
-                                                           str_response.end()));
-        }
-      };
+  auto str_response = api_response.response.str();
+  if (api_response.status != olp::http::HttpStatusCode::OK) {
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/src/generated/api/BlobApi.cpp]:
```suggestion
  if (api_response.status != http::HttpStatusCode::OK) {
```

File: olp-cpp-sdk-dataservice-read/src/generated/api/BlobApi.h [hunk 5]
```diff
@@ -56,15 +57,16 @@ class BlobApi {
    * only supports a single byte range. The range parameter can also be
    * specified as a query parameter, i.e. range=bytes=10-. For volatile layers
    * use the pagination links returned in the response body.
-   * @param callback A callback function to invoke with the data response.
+   * @param context A CancellationContext, which can be used to cancel request.
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/src/generated/api/BlobApi.h]:
```suggestion
   * @param context A CancellationContext, which can be used to cancel the pending request.
```

File: olp-cpp-sdk-dataservice-read/src/generated/api/BlobApi.h [hunk 6]
```diff
@@ -56,15 +57,16 @@ class BlobApi {
...
-  static client::CancellationToken GetBlob(
-      const client::OlpClient& client, const std::string& layerId,
-      const std::string& dataHandle, boost::optional<std::string> billingTag,
-      boost::optional<std::string> range, const DataResponseCallback& callback);
+  static DataResponse GetBlob(const client::OlpClient& client,
+                              const std::string& layer_id,
+                              const std::string& data_handle,
+                              boost::optional<std::string> billing_tag,
+                              boost::optional<std::string> range,
+                              client::CancellationContext context);
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/src/generated/api/BlobApi.h]: I wonder if we really need to pass this by copy. Internally everything is sync and we only pass it to OlpClient which will take care that it copies it when necessary.

Comment (andescu) [reply olp-cpp-sdk-dataservice-read/src/generated/api/BlobApi.h]: I mean the `CancellationContext`, just to be clear. All other parameters should be ok by copy.

File: olp-cpp-sdk-dataservice-read/src/repositories/DataRepository.cpp [hunk 7]
```diff
@@ -134,63 +134,16 @@ DataResponse DataRepository::GetBlobData(
...
-  if (cancellation_context.IsCancelled()) {
-    // We can't use blob response here because it could potentially be
-    // uninitialized.
-    return ApiError(ErrorCode::Cancelled, "Operation cancelled.");
+  if (service == kBlobService) {
+    blob_response = BlobApi::GetBlob(
+        blob_api.GetResult(), layer, data_handle.value(),
+        data_request.GetBillingTag(), boost::none, cancellation_context);
+  } else {
+    blob_response = BlobApi::GetBlob(
```
Comment (mykhailo-kuchma) [inline olp-cpp-sdk-dataservice-read/src/repositories/DataRepository.cpp]: Should be VolatileBlob::GetBlob

Comment (mykhailo-kuchma) [reply olp-cpp-sdk-dataservice-read/src/repositories/DataRepository.cpp]: I wonder why it was not caught by the tests. We should add one.

**Pull request merged**